workspace/codespaces/codespaces.py

```diff
@@ -0,0 +1,96 @@
+import collections
+import datetime
+import json
+import os
+
+import requests
+
+from workspace.utils import blocks
+
+
+URL_PATTERN = "https://api.github.com/orgs/{org}/codespaces"
+# The following can be a classic PAT with the admin:org scope or a fine-grained token
+# with "Codespaces" repository permissions set to "read" and "Organization codespaces"
+# organization permissions set to "read". For more information, see:
+# https://docs.github.com/en/rest/codespaces/organizations?apiVersion=2022-11-28#list-codespaces-for-the-organization
+TOKEN = os.environ["CODESPACES_GITHUB_API_TOKEN"]
+HEADERS = {
+    "Accept": "application/vnd.github+json",
+    "X-GitHub-Api-Version": "2022-11-28",
+}
+
+
+Codespace = collections.namedtuple(
+    "Codespace",
+    ["last_used_days_ago", "owner", "name", "has_unpushed", "has_uncommitted"],
+)
+
+
+def fetch(url, key):
+    """
+    Recursively fetch all records from the GitHub REST API endpoint given by `url`.
+    `key` is the property under which the records are stored.
+    """
+
+    def set_bearer_auth(request):
+        request.headers["Authorization"] = f"Bearer {TOKEN}"
+        return request
+
+    def fetch_one_page(page_url):
+        response = requests.get(page_url, auth=set_bearer_auth, headers=HEADERS)
+        response.raise_for_status()
+        yield from response.json().get(key)
+        if "next" in response.links:
+            yield from fetch_one_page(response.links["next"]["url"])
+
+    yield from fetch_one_page(url)
+
+
+def get_codespace(record):
+    now = datetime.datetime.now(datetime.timezone.utc)
+    last_used_at = datetime.datetime.fromisoformat(record["last_used_at"])
+    last_used_days_ago = (now - last_used_at).days
+    owner = record["owner"]["login"]
+    name = record["name"]
+    has_unpushed = record["git_status"]["has_unpushed_changes"] == "true"
+    has_uncommitted = record["git_status"]["has_uncommitted_changes"] == "true"
+    return Codespace(last_used_days_ago, owner, name, has_unpushed, has_uncommitted)
+
+
+def is_at_risk(codespace, threshold_in_days):
+    return codespace.last_used_days_ago >= threshold_in_days
+
+
+def main():
+    org = "opensafely"
+    threshold_in_days = 20
```

Comment:
How often do we think this command will be automatically run? That determines what the threshold should be to some extent. Is the 20 days calculated based on 28 - 7 - 1? Can we document that?